internal/nightfallconfig/nightfall_config.go

```diff
@@ -32,28 +39,40 @@ type Config struct {
 	FileExclusionList          []string
 }
 
-// GetNightfallConfigFile loads nightfall config from file
-func GetNightfallConfigFile(workspacePath, fileName string) (*NightfallConfigFileStructure, error) {
+// GetNightfallConfigFile loads nightfall config from file, returns default if missing/invalid
+func GetNightfallConfigFile(workspacePath, fileName string, logger logger.Logger) *NightfallConfigFileStructure {
+	defaultNightfallConfig := &NightfallConfigFileStructure{
+		Detectors:         []*nightfallAPI.Detector{&apiKeyDetector, &cryptoKeyDetector},
+		MaxNumberRoutines: DefaultMaxNumberRoutines,
+	}
 	nightfallConfigFile, err := os.Open(path.Join(workspacePath, fileName))
 	if err != nil {
-		return nil, err
+		logger.Warning(fmt.Sprintf("Error opening nightfall config: %s", err.Error()))
+		logger.Info(defaultDetectorsInfoMessage)
+		return defaultNightfallConfig
 	}
 	defer nightfallConfigFile.Close()
 	byteValue, err := ioutil.ReadAll(nightfallConfigFile)
 	if err != nil {
-		return nil, err
+		logger.Warning(fmt.Sprintf("Error reading nightfall config: %s", err.Error()))
+		logger.Info(defaultDetectorsInfoMessage)
+		return defaultNightfallConfig
 	}
 	var nightfallConfig NightfallConfigFileStructure
 	err = json.Unmarshal(byteValue, &nightfallConfig)
 	if err != nil {
-		return nil, err
+		logger.Warning("Invalid nightfall config")
```

Comment:
i'm not sure we want to use the default conditions here - if someone misconfigures their config (say they want to scan for SSN and CC numbers), and we just scan all of their code w/ a different condition set I think it could be confusing (most people won't be reading their Action/Circle logs for this stuff).

What do you think? Should we throw an error and force them to fix it or delete it, or should we just move ahead w/ scanning using our default detectors?

Reply:
Ok that makes sense - since they're already explicitly including a nightfall config, we should expect it to be valid (and error otherwise)